modules/repository/repo.go

```diff
@@ -71,7 +71,15 @@ func MigrateRepositoryGitData(ctx context.Context, u *user_model.User,
 		return repo, fmt.Errorf("Failed to remove %s: %v", repoPath, err)
 	}
 
-	if err = git.CloneWithContext(ctx, opts.CloneAddr, repoPath, git.CloneRepoOptions{
+	cargs := make([]string, len(git.GlobalCommandArgs))
+	copy(cargs, git.GlobalCommandArgs)
+
+	if opts.UseGitCredentials {
```

Comment:
UseGitCredentials variable is not propagated correctly and cargs never get credential.helper appended.